File: java/exercises/inheritance/TestProgression.java
/** Test program for the progession hierarchy. */
public class TestProgression {
  public static void main(String[] args){
    Progression prog;
    // test default progression
    System.out.println("Default progression");
    prog = new Progression();
    prog.printProgression(10);

    // Test arithmetic progression
    System.out.println("AP with default increment");
    prog = new ArithmeticProgression();
    prog.printProgression(10);
    System.out.println("AP with increment 5");
    prog = new ArithmeticProgression(5);
    prog.printProgression(10);
    System.out.println("AP with start 2 increment 5");
    prog = new ArithmeticProgression(5, 2);
    prog.printProgression(10);

    // Test geometric progression
    System.out.println("GP with default base");
    prog = new GeometricProgression();
    prog.printProgression(10);
    System.out.println("GP with base 3");
    prog = new GeometricProgression(3);
    prog.printProgression(10);
    System.out.println("GP with base 3 and start 4");
    prog = new GeometricProgression(3, 4);
    prog.printProgression(10);

    // Test Fibonacci progression
    System.out.println("Traditional FP");
    prog = new FibonacciProgression();
    prog.printProgression(10);
    System.out.println("Custom FP with first 4 and second 6");
    prog = new FibonacciProgression(4, 6);
    prog.printProgression(10);

  }
}

File: python/exercises/inheritance/test_progression.py
from  Progression import Progression
from GeometricProgression import GeometricProgression
from ArithmeticProgression import ArithmeticProgression
from FibonacciProgression import FibonacciProgression

if __name__ == '__main__':
    # test default progression
    print("Default progression")
    Progression().print_progression(10)

    # test arithmetic progression
    print("Arithmetic progression with default increment 1")
    ArithmeticProgression().print_progression(10)

    print("Arithmetic progression with increment 5")
    ArithmeticProgression(increment=5).print_progression(10)

    print("Arithmetic progression with start 10 increment 3")
    ArithmeticProgression(increment=3, start=10).print_progression(10)

    # test geometric progression
    print("Geometric Progression with default base:")
    GeometricProgression().print_progression(10)

    print("Geomertic Progression with base 3")
    GeometricProgression(3).print_progression(10)

    # test fibonacci progression
    print("Fibonacci progression with default first and second terms")
    FibonacciProgression().print_progression(10)

    print("Fibonacci progression with first 5 second 10")
    FibonacciProgression(5, 10).print_progression(10)
